lib/iris/util.py

```diff
@@ -696,22 +696,39 @@ def _build_full_slice_given_keys(keys, ndim):
     return full_slice
 
 
-def _slice_data_with_keys(data, keys, n_parent_dims):
-    # Combine _build_full_slice_given_keys and column_slices_generator to index
-    # any array-like object with arbitrary keys using an orthogonal indexing
-    # interpretation.
-    # This means that both numpy and lazy data will index in the same way.
-    # Return a dimension map, and the indexed data.
-    #
-    # NOTE: by using slicing on only one index at a time, this also generally
-    # avoids copying the data, except when a key contains a list of indices.
-    full_slice = _build_full_slice_given_keys(keys, n_parent_dims)
-
-    dims_mapping, slices_iter = column_slices_generator(full_slice,
-                                                        n_parent_dims)
+def _slice_data_with_keys(data, keys):
+    """
+
+    Index an array-like object with orthogonal keys.
+
+    This enforces an orthogonal interpretation of indexing, which means that
+    both 'real' (numpy) arrays and other array-likes index in the same way,
+    instead of numpy arrays doing 'fancy indexing'.
+
```

Comment:
You could do with a section 'Args:' in the docstring that describes what each of the incoming arguments does.